html/dom/reflection.js

```diff
@@ -501,45 +583,82 @@ ReflectionTests.typeMap = {
      * "Except where otherwise specified, if an IDL attribute that is a
      * floating point number type (double) is assigned an Infinity or
      * Not-a-Number (NaN) value, a NOT_SUPPORTED_ERR exception must be raised."
-     *
-     * TODO: Implement the actual algorithm so we can run lots more tests.  For
-     * now we're stuck with manually setting up expected values.  Of course,
-     * a lot of care has to be taken in checking equality for floats . . .
-     * maybe we should have some tolerance for comparing them.  I'm not even
-     * sure whether setting the content attribute to 0 should return 0.0 or
-     * -0.0 (the former, I hope).
      */
     "double": {
         "jsType": "number",
         "defaultVal": 0.0,
         "domTests": [minInt - 1, minInt, -36, -1, 0, 1, maxInt,
-            maxInt + 1, maxUnsigned, maxUnsigned + 1, "",
+            maxInt + 1, maxUnsigned, maxUnsigned + 1, "", "-", "+",
             "\u00097", "\u000B7", "\u000C7", "\u00207", "\u00A07", "\uFEFF7",
             "\u000A7", "\u000D7", "\u20287", "\u20297", "\u16807", "\u180E7",
             "\u20007", "\u20017", "\u20027", "\u20037", "\u20047", "\u20057",
-            "\u20067", "\u20077", "\u20087", "\u20097", "\u200A7", "\u202F7",
-            "\u30007",
+            "\u20067", "\u20077", "\u20087", "\u20097", "\u200A7", "\u202F7", "\u30007",
+            "\t\u000B7", "\n\u000B7","\f\u000B7", "\r\u000B7", "\x20\u000B7", "7\u000B",
             " " + binaryString + " foo ", undefined, 1.5, "5%", "+100", ".5", true, false,
+            "1.", "1e2", "1e+2", "1e-2", "1E2", "1E+2", "1E-2", "1.e2", "1.0e2",
+            "1. 1", "1 .1", "1. e2", "1 .e2", "1 e2", "1e 2", "1e -2", "1e- 2",
```

Comment:
Maybe add strings that parse to Infinity and -Infinity, e.g. `"1.8e308"` and `"-1.8e308"`